4_predict_plots.R

```diff
@@ -0,0 +1,1845 @@
+source("4_predict/src/select_features.R")
+source("4_predict/src/train_models.R")
+source("4_predict/src/plot_diagnostics.R")
+source("1_fetch/src/generate_credentials.R")
+
+#Predict phase
+p4_plot_targets_list <- list(
+  # Visualize Model Diagnostics:
+  # see note at top of 1_fetch.R re why these are 'local' file targets
+  
+  # Boruta screening
+  # tar_target(p4_Boruta_static_png,
+  #            plot_Boruta(p4_Boruta_static$brf_All,
+  #                        model_name = 'daily_SC_RF',
+  #                        out_dir = '4_predict/out/Boruta'),
+  #            deployment = 'main',
+  #            format = 'file'
+  # ),
+  
+  # RF variable importance plot 
+  #Should add error bars over X random seeds
+  tar_target(p4_vip_static_png,
+             plot_vip(RF_model = p4_train_RF_static$best_fit,
+                      model_name = 'daily_SC_RF_static',
+                      num_features = 20,
+                      out_dir = '4_predict/out/random/vip'),
+             deployment = 'main',
+             format = 'file',
+             repository = 'local'
+  ),
+  tar_target(p4_vip_static_dynamic_png,
+             plot_vip(RF_model = p4_train_RF_static_dynamic$best_fit,
+                      model_name = 'daily_SC_RF_static_dynamic',
+                      num_features = 20,
+                      out_dir = '4_predict/out/random/vip'),
+             deployment = 'main',
+             format = 'file',
+             repository = 'local'
+  ),
```

Comment:
It does look like a lot of these targets are repeated across different versions of the RF models and so might be good candidates for leveraging targets' branching capabilities. 

`targets` requires that we map over another target (as opposed to some vector or data frame in the global environment), so we would need to add a target that specifies the model versions. One option would be to compile the different models within a list such that `plot_vip` could be mapped over each element within that list. 

What's not as clear to me without playing with the code is how we would define the `out_dir` for each of those branches. I can try to think more about it, and maybe adding branching in `4_predict_plots.R` is something we could add as an enhancement issue that gets tackled apart from the rest of this PR.

Reply:
What is the `str()` of `p4_train_RF_static_dynamic`?

Reply:
> adding branching in 4_predict_plots.R is something we could add as an enhancement issue that gets tackled apart from the rest of this PR

I agree this is more an enhancement. I can add an issue

Reply:
`p4_train_RF_static_dynamic` is a list that contains a table of model results, the best fit model evaluated on the test dataset, and the model workflow that can be used to predict with a new dataset (e.g., within `predict()`)

Reply:
I wonder if, in a new target, we could add elements to those lists that contain the `model_name` and `out_dir` and then map over each list to populate the arguments in `plot_vip`. I don't think I've tried mapping over a nested list, but I don't see why it wouldn't work...

Reply:
That sounds good! I can add that idea to the enhancement issue